frontend/src/core/functions/FunctionRegistry.ts

```diff
@@ -15,3 +21,13 @@ export const FUNCTIONS_REGISTRY = new DeferredRequestRegistry<
     ...req,
   });
 });
+
+export const PreviewSQLTable = new DeferredRequestRegistry<
```

Comment:
let me know if its worth wrapping `DeferredRequestRegistry` in an LRU cache. happy to write a class for that if you'd want